mvicore-diff/src/main/java/com/badoo/mvicore/ModelWatcher.kt

```diff
@@ -0,0 +1,52 @@
+package com.badoo.mvicore
+
+class ModelWatcher<T> private constructor(
+    private val watchers: List<Watcher<T, Any?>>
+) {
+    private var model: T? = null
+
+    operator fun invoke(value: T) {
+        val state = model
```

Comment:
The local variable name `state` is a bit confusing. ModelWatcher works with `model`, so I think 
```suggestion
        val prevModel = model
```
ModelWatcher works with `model`, so I think that would be better.